src/mailkit.cs

```diff
@@ -0,0 +1,423 @@
+	[Native, Advice ("'MEMessageActionMessageColor' is not available in UIKit on macOS.")]
+	public enum MEMessageActionMessageColor : long
+	{
+		None,
+		Green,
+		Yellow,
+		Orange,
+		Red,
+		Purple,
+		Blue,
+		Gray,
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Native, Advice ("'MEMessageState' is not available in UIKit on macOS.")]
+	public enum MEMessageState : long
+	{
+		Received = 0,
+		Draft = 1,
+		Sending = 2,
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEMessage' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEMessage : NSSecureCoding
+	{
+		[Export ("state", ArgumentSemantic.Assign)]
+		MEMessageState State { get; }
+
+		[Export ("subject")]
+		string Subject { get; }
+
+		[Export ("fromAddress", ArgumentSemantic.Copy)]
+		MEEmailAddress FromAddress { get; }
+
+		[Export ("toAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] ToAddresses { get; }
+
+		[Export ("ccAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] CcAddresses { get; }
+
+		[Export ("bccAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] BccAddresses { get; }
+
+		[Export ("replyToAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] ReplyToAddresses { get; }
+
+		[Export ("allRecipientAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] AllRecipientAddresses { get; }
+
+		[Export ("dateSent", ArgumentSemantic.Copy)]
+		[NullAllowed]
+		NSDate DateSent { get; }
+
+		[Export ("dateReceived", ArgumentSemantic.Copy)]
+		[NullAllowed]
+		NSDate DateReceived { get; }
+
+		[NullAllowed, Export ("headers", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSArray<NSString>> Headers { get; }
+
+		[NullAllowed, Export ("rawData", ArgumentSemantic.Copy)]
+		NSData RawData { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEMessageAction' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEMessageAction : NSSecureCoding
+	{
+		[Static]
+		[Export ("moveToTrashAction")]
+		MEMessageAction MoveToTrash { get; }
+
+		[Static]
+		[Export ("moveToArchiveAction")]
+		MEMessageAction MoveToArchive { get; }
+
+		[Static]
+		[Export ("moveToJunkAction")]
+		MEMessageAction MoveToJunk { get; }
+
+		[Static]
+		[Export ("markAsReadAction")]
+		MEMessageAction MarkAsRead { get; }
+
+		[Static]
+		[Export ("markAsUnreadAction")]
+		MEMessageAction MarkAsUnread { get; }
+
+		[Static]
+		[Export ("flagAction")]
+		MEMessageAction Flag { get; }
+
+		[Static]
+		[Export ("unflagAction")]
+		MEMessageAction Unflag { get; }
+
+		[Static]
+		[Export ("setColorActionWithColor:")]
+		MEMessageAction SetColor (MEMessageActionMessageColor color);
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEMessageActionDecision' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEMessageActionDecision : NSSecureCoding
```

Comment:
Static class? All methods/properties are static and we have no contractor.

Reply:
This can't be a static class heh

Reply:
And it should not, i'm wrong we get it from https://developer.apple.com/documentation/mailkit/memessageactionhandler/3783568-decideactionformessage?changes=_3&language=objc

So we need to be able to instantiate the class.

Reply:
When I made the interface static, I got these errors in the build: https://gist.github.com/tj-devel709/a4500a4835c99831d0d2f7883f92fe5d

Reply:
Oops, sorry didn't refresh before I posted that ^